src/main/java/org/codehaus/plexus/resource/DefaultResourceManager.java

```diff
@@ -42,18 +40,20 @@
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Named;
 
 /**
  * @author <a href="mailto:trygvis@inamo.no">Trygve Laugst&oslash;l</a>
  * @author Jason van Zyl
  * @version $Id$
  */
-@Component( role = ResourceManager.class, instantiationStrategy = "per-lookup" )
+@Named
 public class DefaultResourceManager implements ResourceManager
 {
     private static final Logger LOGGER = LoggerFactory.getLogger( DefaultResourceManager.class );
 
-    @Requirement( role = ResourceLoader.class )
+    @Inject
```

Comment:
While this IS equivalent what plexus did (field injection), please try to make components immutable (fields `final`) and use ctor injection. Is easier to UT as well, among other benefits.